### PR TITLE
Create st2 user during pre install for deb and rpm packages

### PR DESCRIPTION
**Pull request by jjm**
Add `preinst` file to create st2 user if required (Fixes: #31).

### CODE REVIEW

File: debian/preinst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+# preinst script for st2
```
Comment (arm4b) [inline debian/preinst]: ... for `st2chatops`

File: debian/preinst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+create_users() {
+  # create st2 user (services user)
+  (id $ST2_USER 1>/dev/null 2>&1) ||
+    adduser --group --disabled-password --no-create-home --system $ST2_USER
+
+  # make st2 member of st2packs group
+  (getent group $PACKS_GROUP 1>/dev/null 2>&1) || groupadd -r $PACKS_GROUP
+  (groups $ST2_USER 2>/dev/null | grep -q "\b${PACKS_GROUP}\b") ||
+    usermod -a -G $PACKS_GROUP $ST2_USER
```
Comment (arm4b) [inline debian/preinst]: This part could be removed starting from the `# make st2 member of st2packs group` as `st2packs` are not needed for st2chatops.

If `st2` package is installed on the same machine, - it'll add it anyways.

**Review by arm4b — CHANGES_REQUESTED** [review]
Great thing, much appreciated as it fixes things in several repos :+1:

Needs similar rules for `rpm` as well. You know where to find them ;)

File: debian/preinst (hunk 1 above)
Comment (jjm) [reply debian/preinst]: fixed.

File: debian/preinst (hunk 2 above)
Comment (jjm) [reply debian/preinst]: 👍

**Comment by jjm** [comment]
@armab Sure I'll make the change for rpm too 😱

**Comment by jjm** [comment]
@armab That should be your comments addressed now.

**Review by arm4b — APPROVED** [review]
Looks good :+1: 

Thanks for contribution!

**Comment by arm4b** [comment]
I'll test the packages manually and merge it soon.

**Comment by arm4b** [comment]
I've tried the packages on both Ubuntu16 and CentOS7 and the `s2` user:group gets created.

**Pull request merged**